doc/source/workflows/doc_code/wait_for_event_http.py

```diff
@@ -0,0 +1,53 @@
+import ray
+from ray import workflow, serve
+from ray.workflow.http_event_provider import HTTPEventProvider, HTTPListener
+
+from time import sleep
+import requests
+
+ray.init(storage="/tmp/ray/workflow/data")
+# Start a Ray Serve instance. This will automatically start
+# or connect to an existing Ray cluster.
+serve.start(detached=True)
+HTTPEventProvider.deploy()
+
+# fmt: off
+# __wait_for_event_begin__
+# File name: wait_for_event_http.py
+# Create a task waiting for an http event with a JSON message.
+# The JSON message is expected to have an event_key field
+# and an event_payload field.
+
+event_task = workflow.wait_for_event(HTTPListener, event_key="my_event_key")
+
+obj_ref = workflow.run_async(event_task, workflow_id="workflow_receive_event_by_http")
+
+# __wait_for_event_end__
+
+# fmt: off
+# __submit_event_begin__
+# File name: wait_for_event_http.py
+
+
+def send_event():
+    resp = requests.post(
+        "http://127.0.0.1:8000/event/send_event/"
+        + "workflow_receive_event_by_http",
+        json={"event_key": "my_event_key", "event_payload": "my_event_message"},
+    )
+    return resp
+
+
+# repeat send_event() until the returned status code is not 404
+while True:
+    res = send_event()
+    if res.status_code == 404:
+        sleep(0.5)
+    else:
+        break
```

Comment:
This code looks not very good. how about the error errors? i think the user might just try to run this code.
Also the audience might think why a while loop here? We should add comment to explain this.

Reply:
Client code updated without the loop. Please review.